features/proposed/FxA-76-metrics-opt-out/README.md

```diff
@@ -0,0 +1,204 @@
+from Firefox Accounts and related services.
+
+As a user, I want Firefox Accounts to
+automatically respect the data choices
+that I've already expressed in my local Firefox settings.
+
+
+## Details
+
+Firefox users can make choices about the data they share with Mozilla
+by using the following UI in Firefox:
+
+![](browser_data_choices.png)
+
+Which toggle the following options in about:config:
+
+* datareporting.healthreport.uploadEnabled
+* toolkit.telemetry.enabled
+
+Which in turn control whether
+the browser will collected and submit
+usage and performance metrics.
+
+Firefox Accounts should provide a similar level of control
+over its metrics data collection,
+and should automatically respect the user's data choices
+that they've already made in the browser.
+We'll do this by adding
+a single opt-out preference
+to control metrics gathering behaviour.
+
+
+### Metrics opt-out preference
+
+We will add an additional field
+called "metrics\_optout"
+to the user's profile data,
+which will have three possible states:
+
+* `null`: the user has not made an explicit choice to disable metrics.
+* `false`: the user has explicitly chosen to allow metrics collection.
+* `true`: the user has explicitly chosen to disable metrics collection.
+
+The default will be `null`,
+and metrics-gathering will be assumed enabled by default,
+analogous to the way Health Report
+is enabled by default in Firefox.
+
+The preference will be managed by fxa-profile-server
+in the same manner as existing profile data elements,
+using the following new API endpoints:
+
+* `GET /v1/metrics_optout`
+* `POST /v1/metrics_optout`
+
+The preference will be readable by any relier
+granted the "profile:metrics\_optout" scope,
+which will only be available
+to trusted internal reliers.
+Mozilla properties that rely on Firefox Accounts
+should consider respecting the user's preference,
+although we will not make it a hard requirement
+for becoming a relier.
+
+### Effects of the preference
+
+For users with `metrics_optout=true`,
+we will disable non-essential metrics collection by:
+
+* Disabling submission of in-page metrics from the content-server
+  (specifically, data currently submitted to the `post-metrics`
+  and `get-metrics-errors` endpoints).
+* Ensuring they are excluded from
+  (or failing that, are in the control group for)
+  any A/B experiments
+* Communicating the preference to iframe embedders
+  such as the first-run page,
+  so that they too can respect it.
+
+The preference does *not* opt the user out of metrics gathered
+as a side-effect of operations essential to the service,
+such as aggregated server log metrics
+or the server-side "activity events" metrics stream.
+
+XXX TODO: does this cost us a round-trip to fetch the preference
+or can we inline it with work we're already doing?
+
+### Exposing the preference
+
+XXX TODO: what sort of UI treatment do we want
+for the initial version of this, if any?
```

Comment:
At least initially, since FxA largely supports Sync in Firefox, I would propose that the UI for changing your data preferences is just the one currently in the browser and we just respect the choices you made there.